cmake/FindTBB.cmake

```diff
@@ -257,6 +257,23 @@ FOREACH ( COMPONENT ${TBB_FIND_COMPONENTS} )
     PATHS ${_TBB_LIBRARYDIR_SEARCH_DIRS}
     PATH_SUFFIXES ${TBB_PATH_SUFFIXES}
     )
+
+  # On Unix, TBB sometimes uses linker scripts instead of symlinks, so parse the linker script
+  # and correct the library name if so
+  IF ( UNIX AND EXISTS ${Tbb_${COMPONENT}_LIBRARY} )
+    # Ignore files where the first four bytes equals the ELF magic number
+    FILE ( READ ${Tbb_${COMPONENT}_LIBRARY} Tbb_${COMPONENT}_HEX OFFSET 0 LIMIT 4 HEX )
+    IF ( NOT ${Tbb_${COMPONENT}_HEX} STREQUAL "7f454c46" )
+      # Read the first 1024 bytes of the library and match against an "INPUT (file)" regex
+      FILE ( READ ${Tbb_${COMPONENT}_LIBRARY} Tbb_${COMPONENT}_ASCII OFFSET 0 LIMIT 1024 )
+      IF ( "${Tbb_${COMPONENT}_ASCII}" MATCHES "INPUT \\(([^(]+)\\)")
+        # Extract the directory and apply the matched text (in brackets)
+        GET_FILENAME_COMPONENT ( Tbb_${COMPONENT}_DIR "${Tbb_${COMPONENT}_LIBRARY}" DIRECTORY )
+        SET ( Tbb_${COMPONENT}_LIBRARY "${Tbb_${COMPONENT}_DIR}/${CMAKE_MATCH_1}" )
+      ENDIF ()
+    ENDIF ()
+  ENDIF ()
+
```

Comment:
I've tried to resolve this already by simply including the directory. See like 318:

```
  # CMake sometimes struggles to follow TBB's imported target, as libtbb.so
  # is usually just a file which defines "INPUT (libtbb.so.2)". As it's not
  # a symlink, we need to include this directory manually incase it's not on
  # the environment library path
  LINK_DIRECTORIES ( ${Tbb_LIBRARY_DIRS} )
```

This is definitely better if it's a concrete solution? AS it will avoid runtime warnings from cmake if we can remove the above.

Reply:
Ah, I didn't spot this, this didn't seem to work for me. The method I'm proposing is a bit more involved, but I think probably a better solution because it actually interprets the linker script. Knowing that Houdini ships with this library linking by default on Linux, I think it's something we should explicitly handle.

Reply:
We should be able to remove that link_directories command now, I'll have a test my end

Reply:
Removed